gotham/src/router/non_match.rs

```diff
@@ -86,7 +102,23 @@ impl RouteNonMatch {
     /// which wrap other `RouteMatcher` instances. See the `Node::select_route` implementation (in
     /// `gotham::router::tree`) for an example.
     pub fn union(self, other: RouteNonMatch) -> RouteNonMatch {
-        let status = higher_precedence_status(self.status, other.status);
+        let status = match (self.status, other.status) {
```

Comment:
I guess this can't just be an `impl PartialOrd` because `StatusCode` originates from Hyper? If we ever want to add any more gotham specific functionality, it could be worth a newtype, at which point this would be simplified.

Reply:
The point of this PR is to use different compare logic for union vs intersection, so `impl PartialOrd` wouldn't do it either way.

Reply:
Oh, my eyes glazed over the fact that they are different. LGTM!